### DISCUSSION TITLE
Plugin SQL Enterprise...

### DISCUSSION THREAD

**Discussion by patrikloren** [pawelsalawa/sqlitestudio#4197] [Q&A]
Hi, I'm looking for a good formatter for SQL and stumbled on this one when using SQLiteStudio (ver 3.3.3).
I've searched for more information on this one to try to use it in other tools also, such as SSMS or ADS.
I do know it's made by SalSoft but I haven't been able to locate them.
Do they have a homepage or some way of contacting them? Or has someone already wrapped the SqlEnterpriseFormatter.dll to use it??
I do know about some other extensions to SSMS (SSMSBoost, Poor SQL Formatter and a few others) or ADS(I've tried SQL Prompt free extension here and it lacks some good features I'm after) and I can't use SQL Prompt (which is really got but cost some $$).

Grateful for all help!

**Comment by pawelsalawa** [comment]
I wrote enterprise formatter. SalSoft is my company :)

I don't think you will be able to use this formatter outside of SQLiteStudio, because it has many code dependencies from the SQLiteStudio (coreSQLiteStudio.dll). It was written purely for SQLiteStudio.

Also it is not universal. It works with SQLite3 grammar, not with PostgreSQL grammar, nor MSSQL, etc. Sure, for some basic statements, which are common for all databases - it understands the grammar, but each database has its own grammar and the enterprise formatter from SQLiteStudio would understand only that from SQLite.
